gammapy/spectrum/tests/test_extract.py

```diff
@@ -165,3 +167,30 @@ def test_compute_energy_threshold(self, extraction):
         extraction.compute_energy_threshold(method_lo="area_max", area_percent_lo=10)
         actual = extraction.spectrum_observations[0].energy_range[0]
         assert_quantity_allclose(actual, 0.8799225 * u.TeV, rtol=1e-3)
+
+@requires_data()
+def test_extract_cta_1dc_data(caplog):
```

Comment:
This is the first use of the `caplog` fixture and to assert on logging.

OK to put this in like this now, but I'd like to look at this in some more detail if this is the way we want to do it or not. Logging can be difficult because it has global config state, so we need a good strategy and use it consistently for tests.

I'll make a reminder issue for myself now to look into this, and to make an entry in the dev howto for this.

Reply:
Yes `spectrum` does a bit of logging, but this is never really tested. I suppose we need to check for that too.

Reply:
@registerrier - I'd suggest to not add any tests for logging from Gammapy at this point.

We don't have good logging yet, it's something we need to add in the future, but probably only after the code is in better shape, not now.

It's not clear if we want to use std lib `logging`, or rather some structured logging like https://eliot.readthedocs.io/en/stable/ or something for [provenence](https://cta-observatory.github.io/ctapipe/api/ctapipe.core.Provenance.html) like what ctapipe has.

So IMO: OK to add / remove / improve `log` statements in Gammapy where you think it makes sense, but let's not add tests for that now.

Reply:
Keeping this one case or having a few logging tests is fine. I just meant that we shouldn't add 100s or 1000s of asserts for this.